### PR TITLE
Codegen testkit docs

### PR DESCRIPTION
**Pull request by patriknw**
Some adjustments to https://github.com/lightbend/akkaserverless-java-sdk/pull/334 so that we can merge it.

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
Will need updates right away but let's merge.

**Comment by patriknw** [comment]
Now I'm uncertain about the attribution if we squash this, let me change author to cisco to all commits (can't rebase because it has been merged)

**Comment by raboof** [comment]
I thought it takes "whoever opened the PR", so changing the author of the commits wouldn't be sufficient

**Comment by patriknw** [comment]
> I thought it takes "whoever opened the PR", so changing the author of the commits wouldn't be sufficient

crap, I guess I should be able to do a merge squash locally. Do you know the command for that?

**Comment by raboof** [comment]
I think I would:

```
$ git checkout codegen-testkit-docs
$ git rebase -i 8a30bb2^
(mark all commits except the first one as 'fixup')
```

and then do a 'regular' merge of that manually-squashed commit.

**Comment by patriknw** [comment]
I squashed this, and we can MERGE (no squash), which should keep the attribution to cisco

File: docs/src/modules/java/examples/testkit/java-eventsourced-shopping-cart/intarget/generated-test-sources/akkaserverless/com/example/shoppingcart/domain/ShoppingCartTestKit.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import com.akkaserverless.javasdk.testkit.Result;
+import com.akkaserverless.javasdk.testkit.impl.AkkaServerlessTestKitHelper;
+import com.example.shoppingcart.ShoppingCartApi;
+import com.google.protobuf.Empty;
+import java.util.List;
+
+public class ShoppingCartTestKit {
+
+
+  public ShoppingCartTestKit(ShoppingCart entity) // <1>
```
Comment (patriknw) [inline docs/src/modules/java/examples/testkit/java-eventsourced-shopping-cart/intarget/generated-test-sources/akkaserverless/com/example/shoppingcart/domain/ShoppingCartTestKit.java:13]: We should document this without having this fake class.

File: docs/Makefile [hunk 2]
```diff
@@ -57,6 +58,10 @@ examples:
 	rsync -a --exclude-from=.examplesignore ../samples/doc-snippets/ "${managed_examples}/doc-snippets/"
 	rsync -a --exclude-from=.examplesignore ../samples/first-service/ "${managed_examples}/first-service/"
 
+testkits:
+	mkdir -p "${managed_examples}"
+	rsync -a --exclude-from=.examplesignore ../testkit/src/main/java/ "${managed_examples}/testkit/"
```
Comment (patriknw) [inline docs/Makefile:63]: this is probably not needed

**Review by patriknw — COMMENTED** [review]
some comments for later follow up

**Pull request merged**